packit/upstream.py

```diff
@@ -405,14 +412,17 @@ def _create_archive_using_default_way(self, dir_name, env, version) -> str:
 
         :return: name of the archive
         """
-        archive_extension = self.get_archive_extension(dir_name, version)
-        if archive_extension not in COMMON_ARCHIVE_EXTENSIONS:
-            raise PackitException(
-                "The target archive doesn't use a common extension ({}), "
-                "git archive can't be used. Please provide your own script "
-                "for archive creation.".format(", ".join(COMMON_ARCHIVE_EXTENSIONS))
-            )
-        archive_name = f"{dir_name}{archive_extension}"
+        if self.package_config.archive_name:
+            archive_name = self.package_config.archive_name.format(version=version)
+        else:
+            archive_extension = self.get_archive_extension(dir_name, version)
+            if archive_extension not in COMMON_ARCHIVE_EXTENSIONS:
```

Comment:
I agree with Hunor: since we now change `Source` and `%setup` lines, we no longer need this.

Reply:
I don't agree with myself :-P , or at least I've looked into it, and I've learnt that the archive format might matter after all:

1. `%setup` has support for only a few archive formats.

2. some packages might chose not to use `%setup` in `%pre`, but directly call an archive tool (unzip, tar).

Due to the above we might want to make sure that Packit creates an archive in the format that the original spec file expects.

To do this we will need to explicitly check and remember the extension of the original archive (in some better more explicit way than the current approach) and use it when calling `git archive`.

We will also need to make sure, that `git` has all the archive formats [configured](https://git-scm.com/docs/git-archive#_configuration).

Note, that the above is an addition to the discussion we were having, not a request for change in this PR :-)